src/libtest/libtest-utils/src/test.rs

```diff
@@ -0,0 +1,85 @@
+//! Tests utilities
+
+use crate::{format::Padding, ShouldPanic};
+use std::{borrow::Cow, fmt};
+
+/// Test description.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct Desc {
```

Comment:
@djrenren 

> Hmmm the *Desc structs still seem to be tied to the libtest types. Currently a Desc contains a functional test 

So not really. Like you can see here, the `Desc` struct only contains the test name, whether it is ignored, whether it should panic, or is it allowed to fail.